src/app.js

```diff
@@ -0,0 +1,103 @@
+// Ik had eerst beide delen van de opdracht in één bestand en repository staan, zodat het een website met twee pagina's was, maar dat werd toch wat onoverzichtelijk en foutgevoelig, dus later heb ik dit toch gescheiden. Uiteraard is deel 1 ook nog terug te vinden op GitHub.
+
+import axios from 'axios';
+
+// Global variables
+
+// Referentie weergegeven zoekresultaat:
+const countryInfoContainer = document.getElementById('country-info-container');
+// Referentie error message:
+const errorMessageContainer = document.getElementById('error-message');
+// Referenties formulier en zoekveld
+const searchQueryField = document.getElementById('search-query-field');
+const searchCountryForm = document.getElementById('search-country-form');
+
+// Deze functie haalt de gevraagde gegevens op uit de API
+async function fetchCountryDetails(name) {
+    // Verwijdert het eventuele voorgaande zoekresultaat
+    countryInfoContainer.innerHTML = '';
+    // Verwijdert eventuele error message van vorige zoekopdracht
+    errorMessageContainer.innerHTML = '';
+
+    try {
+        const response = await axios.get(
+            `https://restcountries.com/v2/name/${name}`
+        );
+        const country = response.data[0]; // 1 resultaat weergeven
+        console.log(country);
+        showCountry(country); // Aanroepen van de functie hieronder, die de resultaten op de pagina zet
+    } catch (e) {
+        // Dit wordt weergegeven als de zoekopdracht niet wordt herkend
+        console.error(e);
+        errorMessageContainer.innerHTML = `
+        <p class="error-message">Country not found, try again.</p>
+        `;
+    }
+}
+
+// Onderstaande functie geeft de informatie over het opgevraagde land weer op de pagina; wordt aangeroepen in try-blok hierboven, waar het country-object beschikbaar is
+
+function showCountry(country) {
```

Comment:
```suggestion
function showCountry({name, flag, subregion, population, capital, currencies, languages}) {
```
Dan hoef je niet overal `country.` vóór te zetten 😉